### ISSUE TITLE
why not use pvportmalloc instead of malloc?

### ISSUE THREAD

**Issue by Yangnaan** [michaelbecker/freertos-addons#59]
malloc and free from stdlib are not thread safe functions

**Comment by michaelbecker** [comment]
Correct, sort of. Each port of FreeRTOS + the associated tool chain each interact in varying ways. Sometimes malloc is safe, sometimes it's not. 

But most of the code I wrote is designed to be used _before_ you start FreeRTOS, in which case using pvPortMalloc may behave in undefined ways, because there isn't a FreeRTOS running yet. And more importantly, since most of this is C++ code, the new operator does it's own thing _independent_ of FreeRTOS. In most cases this means a direct call to malloc then subsequent calls to varying constructors. If you are planning on using C++ in an embedded system, you unfortunately need to get intimately familiar with how all of the memory allocations are done, the standard C ones, the C++ ones, and the FreeRTOS ones, as well as knowing how they all interact with each other for your specific port and compiler.
